specs/interop/derivation.md

```diff
@@ -87,17 +91,185 @@ The interop network upgrade timestamp defines the timestamp at which all functio
+
+The `CrossL2Inbox` contract is deployed.
+
+A deposit transaction is derived with the following attributes:
+
+- `from`: `0x4220000000000000000000000000000000000000`
+- `to`: `null`
+- `mint`: `0`
+- `value`: `0`
+- `nonce`: `0`
+- `gasLimit`: `420000`
+- `data`: `0x6080604052348015600e575f80fd5b...` ([full bytecode](../static/bytecode/interop-cross-l2-inbox-deployment.txt))
+- `sourceHash`: `0x6e5e214f73143df8fe6f6054a3ed7eb472d373376458a9c8aecdf23475beb616`,  
+  computed with the "Upgrade-deposited" type, with `intent = "Interop: CrossL2Inbox Deployment"`
+
+This results in the Interop CrossL2Inbox contract being deployed to
+`0x691300f512e48B463C2617b34Eef1A9f82EE7dBf`, to verify:
+
+```bash
+cast compute-address --nonce=0 0x4220000000000000000000000000000000000000
+Computed Address: 0x691300f512e48B463C2617b34Eef1A9f82EE7dBf
+```
+
+Verify `sourceHash`:
+
+```bash
+cast keccak $(cast concat-hex 0x0000000000000000000000000000000000000000000000000000000000000002 $(cast keccak "Interop: CrossL2Inbox Deployment"))
+# 0x6e5e214f73143df8fe6f6054a3ed7eb472d373376458a9c8aecdf23475beb616
+```
+
+Verify `data`:
+
+```bash
+git checkout 71c460ec7c7c05791ddd841b97bcb664a1f0c753
+make build-contracts
+jq -r ".bytecode.object" packages/contracts-bedrock/forge-artifacts/CrossL2Inbox.sol/CrossL2Inbox.json
+```
+
+This transaction MUST deploy a contract with the following code hash  
+`0x0e7d028dd71bac22d1fb28966043c8d35c3232c78b7fb99fd1db112b5b60d9dd`.
+
+To verify the code hash:
+
+```bash
+git checkout 71c460ec7c7c05791ddd841b97bcb664a1f0c753
+make build-contracts
+cast k $(jq -r ".deployedBytecode.object" packages/contracts-bedrock/forge-artifacts/CrossL2Inbox.sol/CrossL2Inbox.json)
+```
+
+### CrossL2Inbox Proxy Update
+
+This transaction updates the CrossL2Inbox Proxy ERC-1967
+implementation slot to point to the new CrossL2Inbox deployment.
+
+A deposit transaction is derived with the following attributes:
+
+- `from`: `0x0000000000000000000000000000000000000000`
+- `to`: `0x4200000000000000000000000000000000000022` (CrossL2Inbox Proxy)
+- `mint`: `0`
+- `value`: `0`
+- `gasLimit`: `50,000`
+- `data`: `0x3659cfe60000000000000000000000004200000000000000000000000000000000000022`
+- `sourceHash`: `0x88c6b48354c367125a59792a93a7b60ad7cd66e516157dbba16558c68a46d3cb`
+  computed with the "Upgrade-deposited" type, with `intent = "Interop: CrossL2Inbox Proxy Update"`
+
+Verify data:
+
+```bash
+cast concat-hex $(cast sig "upgradeTo(address)") $(cast abi-encode "upgradeTo(address)" 0x691300f512e48B463C2617b34Eef1A9f82EE7dBf)
+# 0x3659cfe60000000000000000000000004200000000000000000000000000000000000022
```

Comment:
This gives `0x3659cfe6000000000000000000000000691300f512e48b463c2617b34eef1a9f82ee7dbf`, seems this was called with argument `4200000000000000000000000000000000000022` instead.